src/utils/ScreenReader.ts

```diff
@@ -0,0 +1,281 @@
+  }
+
+  /**
+   * Call this with an event target when a click event occurs. Resolves when speaking is done.
+   */
+  public onClick = async (target?: EventTarget) => {
+    await this.speakEventTarget(target)
+  }
+
+  /**
+   * Call this when a page load occurs. Resolves when speaking is done.
+   */
+  public async onPageLoad(): Promise<void> {
+    this.tts.stop()
+  }
+
+  /**
+   * Call this when the screen reader is enabled. Resolves when speaking is done.
+   */
+  public async onScreenReaderEnabled(): Promise<void> {
+    await this.speak('Screen reader enabled', { now: true })
+  }
+
+  /**
+   * Call this when the screen reader is disabled. Resolves when speaking is done.
+   */
+  public async onScreenReaderDisabled(): Promise<void> {
+    await this.speak('Screen reader disabled', { now: true })
+  }
+
+  /**
+   * Directly triggers speech of text. Resolves when speaking is done.
+   */
+  public async speak(text: string, options: SpeakOptions = {}): Promise<void> {
+    /* istanbul ignore next */
+    if (process.env.NODE_ENV === 'development') {
+      // eslint-disable-next-line no-console
+      console.log(`[ScreenReader] speak(now: ${options.now || false}) ${text}`)
+    }
+    await this.tts.speak(text, options)
+  }
+
+  /**
+   * Directly triggers speech of an element. Resolves when speaking is done.
+   */
+  public async speakNode(node: Node, options?: SpeakOptions): Promise<void> {
+    const description = this.describe(node)
+    if (description) {
+      await this.speak(description, options)
+    }
+  }
+
+  /**
+   * Directly triggers speech of an event target. Resolves when speaking is done.
+   */
+  public async speakEventTarget(
+    target?: EventTarget,
+    { now = true }: SpeakOptions = {}
+  ): Promise<void> {
+    if (target && target instanceof Element) {
+      await this.speakNode(target, { now })
+    }
+  }
+
+  /**
+   * Generates a text string to be spoken for an element.
+   */
+  public describe(node: Node): string | undefined {
+    if (!(node instanceof Text) && !(node instanceof Element)) {
+      return
+    }
+
+    return this.cleanDescription(
+      node instanceof Text
+        ? this.describeText(node)
+        : this.describeElement(node)
+    )
+  }
+
+  private cleanDescription(
+    description: string | undefined
+  ): string | undefined {
+    if (!description) {
+      return
+    }
+    return description
+      .replace(/ {2}/g, ' ')
+      .replace(/^[. ]+/g, '')
+      .replace(/\. +\./g, '.')
+      .replace(/,\./g, '.')
+      .replace(/ +\./g, '.')
+      .replace(/ +,/g, ',')
+      .replace(/\.+/g, '.')
+      .replace(/ +$/g, '')
+      .replace(/^ +/g, '')
+  }
+
+  private describeText(node: Text): string | undefined {
+    return node.textContent || undefined
+  }
+
+  private describeElement(node: Element): string | undefined {
+    if (this.isHidden(node)) {
+      return
+    }
+
+    const terminator = this.isBlockElement(node) ? '.' : ''
```

Comment:
I love this. I think we can safely remove many `aria-label` attributes where the value is the same as the text content plus a period.